realsense2_camera/src/realsense_node_factory.cpp

```diff
@@ -57,22 +67,132 @@ void RealSenseNodeFactory::getDevice(rs2::device_list list)
 		else
 		{
 			bool found = false;
+
+			if (_accel_orientation.empty()) // generate accel orientation vector from camera quaternion
+			{
+				try {
+					geometry_msgs::TransformStamped transformStamped = 
+									_tf_buffer.lookupTransform(_tf_link_name, "base_link", ros::Time(0));
```

Comment:
I would suggest to get base_link also as a param called reference_frame. Though its not likely in short term, we might change the reference frame at some point to something else and its easier to catch it in launch file.

Reply:
Alright. Will make the changes.

Reply:
Done. Code committed.